corehq/apps/app_manager/tests/test_suite_multi_select_case_list.py

```diff
@@ -352,28 +350,25 @@ def test_parent_selects_parent_same_type(self):
         self.set_parent_select(self.m0, self.m2)
 
         self.assert_module_datums(self.m0.id, [
-            ('datum', 'case_id_beneficiary'),
-            ('instance-datum', 'selected_cases')
+            ('datum', 'case_id_beneficiary'),  # m2
+            ('instance-datum', 'selected_cases')  # m0
         ])
         self.assert_module_datums(self.m1.id, [
-            ('datum', 'case_id_beneficiary'),
+            ('datum', 'case_id_beneficiary'),  # From m1, but copied from m2
```

Comment:
As a refresher:
```
# All of these tests use the same app structure:
# m0 is a parent module and its case list is multi-select
# m1 is a child of m0 and uses the same case type, this is where the tests focus
# m2 is a standalone module of the same case type
# m3 is a standalone module of another case type
```
The issue in this test is that `get_select_chain_meta` only returns the select chain for m1, since it doesn't use `parent_select`, but then in `add_parent_datums` we compare that chain against that of its parent, which contains `case_id_beneficiary` from m2 and `selected_cases` from m0.

When these two are aligned, m1's `case_id` is improperly correlated with `m2`'s `case_id_beneficiary` and reassigned that name, even though both cases come from different case lists.  `get_add_case_preloads_case_id_xpath` is pretty naive, and just anticipates `case_id`.